core/contracts/Voting.sol

```diff
@@ -72,8 +73,20 @@ contract Voting is Testable {
     // Maps round numbers to the rounds.
     mapping(uint => Round) private rounds;
 
+    // The set of identifiers the oracle can provide verified prices for.
+    mapping(bytes32 => bool) private supportedIdentifiers;
```

Comment:
What do you think the pros and cons are of extracting all the supported identifier functionality into a base class would be?

Some advantages might be:
- If we were to build a MockOracle, we could probably inherit it there too.
- We could test it separately from the Voting contract.
- If we wanted to add more information about identifiers on chain (such as an IPFS hash that gives feed sources, voting methodology, or voting function), we could do that without adding complexity to the Voting contract itself.

Disadvantages:
- Added complexity.
- Require us to adhere to an API boundary between the identifiers and the voting contract.


I'm not strongly opinionated here, just wanted to hear what you think.

Reply:
The only weird part would be how we handle the roles. We'd probably have to initialize them in Voting.sol and pass the relevant role id to the base class. At that point, I think it might too trivial a piece of functionality to break it out (only 2 functions, one of which is a one liner).

Totally agree that we shouldn't add unnecessary complexity to the Voting contract. Maybe if we do the IPFS thing, we could move the on-chain identifier configs to a separate contract entirely. Thoughts?

Reply:
Agreed. That SGTM.